src/Neo/Ledger/MemoryPool.cs

```diff
@@ -157,14 +159,14 @@ public bool ContainsKey(UInt256 hash)
         /// <param name="hash">The hash of the <see cref="Transaction"/> to get.</param>
         /// <param name="tx">When this method returns, contains the <see cref="Transaction"/> associated with the specified hash, if the hash is found; otherwise, <see langword="null"/>.</param>
         /// <returns><see langword="true"/> if the <see cref="MemoryPool"/> contains a <see cref="Transaction"/> with the specified hash; otherwise, <see langword="false"/>.</returns>
-        public bool TryGetValue(UInt256 hash, out Transaction tx)
+        public bool TryGetValue(UInt256 hash, [MaybeNullWhen(false)] out Transaction? tx)
         {
             _txRwLock.EnterReadLock();
             try
             {
-                bool ret = _unsortedTransactions.TryGetValue(hash, out PoolItem item)
+                var ret = _unsortedTransactions.TryGetValue(hash, out var item)
                            || _unverifiedTransactions.TryGetValue(hash, out item);
-                tx = ret ? item.Tx : null;
+                tx = ret ? item!.Tx : null;
```

Comment:
Isn't this the same as itms?.tx?

Reply:
I agree should be `item?.tx`

Reply:
It will be != null always that ret is true

Reply:
I meant `tx = item?.tx`

Reply:
From looking it again. It will always have the resolve of `_unverifiedTransactions.TryGetValue(hash, out item)`

Reply:
> It will be != null always that ret is true

It should be equal to `item?.tx`, otherwise there are problems in TryGetValue